### PR TITLE
Add support for musl libc

### PR DESCRIPTION
**Pull request by kmaork**
Hello, thanks for this project, I love it and use it quite often 😄 
It'd be great to add support for musl libc. It's a pretty simple change - we only need to add the the [standard musl shared library path](https://wiki.musl-libc.org/guidelines-for-distributions.html) (under "Standard Pathnames") to the libc regex.

Changes:
- Added support for musl
- Added a CI job to test support for musl
- Changed tests to support musl
- On line 268 I added a call to `waitpid`, as just killing the target processes caused them to linger as zombies
- On line 322 I changed the library name from `Makefile` to `no such library` because the musl error message contains the library's absolute path, so it was simpler to just give it a non-existent path. An added benfit of that change is that it makes the test a bit less confusing in my opinion. When I first saw the error message `unexpected injection error message: dlopen failed: Error loading shared library /__w/injector/injector/tests/Makefile: Exec format error` I thought that there was a bug in the tests causing the injection of a wrong file.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -46,3 +46,18 @@ jobs:
       shell: cmd
       run: |
         cd tests && .\test-prog-x64.exe
+
+  tests_on_alpine_x86_64:
+    name: Tests on Alpine (x86_64 and i686)
```
Comment (kubo) [inline .github/workflows/test.yml:51]: Could you remove `and i686`? It isn't tested here.

File: tests/test-prog.c [hunk 2]
```diff
@@ -267,6 +281,9 @@ int main(int argc, char **argv)
     void *handle = NULL;
     int rv = 1;
     int loop_cnt;
+    // Sadly this is not known at compile time, see https://www.openwall.com/lists/musl/2013/03/29/13
+    int is_musl;
```
Comment (kubo) [inline tests/test-prog.c:285]: Could you move the local variable `is_musl` to [here](https://github.com/kubo/injector/blob/23f3db17bcb03e43d845764824eced610c7aa97a/tests/test-prog.c#L156-L159)?
It is used only for linux.

File: tests/test-prog.c [hunk 3]
```diff
@@ -279,7 +296,9 @@ int main(int argc, char **argv)
     if (process_start(&proc, test_target) != 0) {
         return 1;
     }
-    printf("targe process started.\n");
+    is_musl = process_check_module(&proc, "/ld-musl-") == 1;
+    expected_errmsg = is_musl ? MUSL_INJECT_ERRMSG : INJECT_ERRMSG;
```
Comment (kubo) [inline tests/test-prog.c:300]: 1. `process_check_module` doesn't work as you expected. It returns `1` also for glibc processes.

2. Could you move the above two lines after `sleep(1)` below?
  At this point, `test_target` executed by `process_start` may be in the middle of initializing when this process go forward before CPU resources are assigned to the `test_target` process.  After `sleep(1`), `test_target` will be initialized.

File: .github/workflows/test.yml (hunk 1 above)
Comment (kmaork) [reply .github/workflows/test.yml:51]: Mmm right! By the way, can you approve the CI to run?

File: tests/test-prog.c (hunk 3 above)
Comment (kmaork) [reply tests/test-prog.c]: Done

File: tests/test-prog.c (hunk 2 above)
Comment (kmaork) [reply tests/test-prog.c]: Done

**Pull request merged**